src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py

```diff
@@ -3726,6 +3726,68 @@ def test_vm_generic_update(self, resource_group):
 
 
 class VMGalleryImage(ScenarioTest):
+    @ResourceGroupPreparer(location='eastus')
+    def test_shared_gallery(self, resource_group, resource_group_location):
+        self.kwargs.update({
+            'vm': 'vm1',
+            'gallery': self.create_random_name(prefix='gallery_', length=20),
+            'image': 'image1',
+            'version': '1.1.2',
+            'captured': 'managedImage1',
+            'location': resource_group_location,
+            'subId': '0b1f6471-1bf0-4dda-aec3-cb9272f09590',  # share the gallery to tester's subscription, so the tester can get shared galleries
+            'tenantId': '2f4a9838-26b7-47ee-be60-ccc1fdec5953',
+            'sharedSubId': '34a4ab42-0d72-47d9-bd1a-aed207386dac'
```

Comment:
After it is transferred to Yong, these subscriptions and tenants can't be accessed.

Reply:
subId is the subscription of tester. Now it is AzureSDKTest. In this test, I share the gallery of AzureSDKTest to itself. So we can test get operation as the shared side.
tenantId and sharedSubId is just hardcoded random GUID. It will be added to shared object, and then removed. So whatever it is, it will not influence test result.
If you want to live test the code again, you just need to `subId` to your subscription ID

Reply:
Thanks for the information! Could we use `get_subscription_id(self.cli_ctx)` in `azure.cli.core.commands.client_factory` to get subscription id instead of hard code one.

Reply:
I have tried `get_subscription_id(self.cli_ctx)`, but the subscription ID got by this function will be set to all-zero in recording. However, the response of service is the original subscription ID. So the check will fail.

Reply:
Thanks for the try!

Reply:
It is an opportunity to optimize subscription replacer.

Reply:
Hi @Juliehzl, another notice: since the service has not gone to private preview and the sig service need to collaborate will VM later, I am not sure we can still share a gallery to ourselves/random tenantID in the future. You may need to live test again when the service go to private/public preview.